app/Http/Controllers/ProfileController.php

```diff
@@ -135,4 +135,95 @@ public function mobileCarrier(Request $request){
             return (new ErrorResource($res))->response()->setStatusCode(400);
         }
     }
+
+    public function updateProfile(Request $request, Validate $validate){ 
+      
+       $user = $request->user();
+       $input = $request->all();
+       $validationErrors = $validate->validate($request, $this->rules->getUpdateProfileValidatonRules(),[]);
+       if( $validationErrors ){
+           return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+       }
+
+       try{           
+            $private_flags = [];
+            if(isset($input['first_name']))
+                $user->first_name =  $input['first_name']; 
+            if(isset($input['middle_name']))
+                $user->middle_name = $input['middle_name'] ;
+            if(isset($input['last_name']))
```

Comment:
first_name and last_name are required fields. Can we ignore `isset` in this case?